### PR TITLE
extensions: Use bazel build for generating docs rst

### PR DESCRIPTION
**Pull request by phlax**
Commit Message: extensions: Use bazel build for generating docs rst
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: bazel/repositories_extra.bzl [hunk 1]
```diff
@@ -36,6 +36,11 @@ def _python_deps():
         # release_date = "2019-02-23"
         # use_category = ["test"],
     )
+    pip_install(
+        name = "docs_pip3",
+        requirements = "@envoy//tools/docs:requirements.txt",
+        extra_pip_args = ["--require-hashes"],
+    )
```
Comment (moderation) [inline bazel/repositories_extra.bzl:43]: What are your thoughts on capturing metadata for the Python deps? I added this data as comments with an example at https://github.com/envoyproxy/envoy/blob/main/bazel/repositories_extra.bzl#L26-L37. Some issues:
- some new deps don't have this data
- no automated checkers
- not machine readable
- duplicitive

Comment (phlax) [reply bazel/repositories_extra.bzl:43]: yep - mostly the issues - ie hard to maintain and in some cases it just duplicates what is there in the egg

Comment (moderation) [reply bazel/repositories_extra.bzl:43]: Thinking maybe we remove the existing comment metadata until we think we need a better solution and agree on a way to do it? (doesn't have to be in this PR which looks like it is on its way to passing CI)

Comment (phlax) [reply bazel/repositories_extra.bzl:43]: sure - im happy to follow that up after

**Review by mattklein123 — APPROVED** [review]
Nice

**Pull request merged**